src/test/itimer/test_itimer.rs

```diff
@@ -0,0 +1,330 @@
+
+    // Sleep for 50 ms.
+    std::thread::sleep(std::time::Duration::from_millis(50));
+    // Shouldn't have fired yet.
+    ensure_ord!(SIGNAL_CTR.load(Ordering::Relaxed), ==, 0);
+    // Should be somewhere between 0 and 100 ms left.
+    let val = getitimer(libc::ITIMER_REAL)?;
+    ensure_ord!(val.value, >, TimeVal::zero());
+    ensure_ord!(val.value, <, TimeVal::milliseconds(100));
+    // Interval should still be 0.
+    ensure_ord!(val.interval, ==, TimeVal::zero());
+
+    // Sleep another 100 ms, which should put us at about 150ms since setting the timer.
+    std::thread::sleep(std::time::Duration::from_millis(100));
+
+    // Should have fired exactly once.
+    ensure_ord!(SIGNAL_CTR.load(Ordering::Relaxed), ==, 1);
+
+    // Timer should no longer be enabled.
+    ensure_ord!(getitimer(libc::ITIMER_REAL)?, ==, ITimer{value: TimeVal::zero(), interval: TimeVal::zero()});
+
+    // Likewise it shouldn't go off again after sleeping again.
+    std::thread::sleep(std::time::Duration::from_millis(150));
+    ensure_ord!(SIGNAL_CTR.load(Ordering::Relaxed), ==, 1);
+    Ok(())
+}
+
+fn test_interval() -> anyhow::Result<()> {
+    reset()?;
+
+    // 100 ms
+    let it_value = libc::timeval {
+        tv_sec: 0,
+        tv_usec: 100_000,
+    };
+    let it_interval = libc::timeval {
+        tv_sec: 0,
+        tv_usec: 100_000,
+    };
+    setitimer(
+        libc::ITIMER_REAL,
+        &libc::itimerval {
+            it_value,
+            it_interval,
+        },
+    )?;
+
+    // Sleep for 150 ms.
+    std::thread::sleep(std::time::Duration::from_millis(150));
+    // Should have fired once.
+    ensure_ord!(SIGNAL_CTR.load(Ordering::Relaxed), ==, 1);
+    // Should be somewhere between 0 and 100 ms left.
+    let val = getitimer(libc::ITIMER_REAL)?;
+    ensure_ord!(val.value, >, TimeVal::zero());
+    ensure_ord!(val.value, <, TimeVal::milliseconds(100));
+    // Interval should still be 100ms.
+    ensure_ord!(val.interval, ==, TimeVal::milliseconds(100));
+
+    // Sleep another 100 ms, which should put us at about 250ms since setting the timer.
+    std::thread::sleep(std::time::Duration::from_millis(100));
+
+    // Should have fired again.
+    ensure_ord!(SIGNAL_CTR.load(Ordering::Relaxed), ==, 2);
+    Ok(())
+}
+
+fn test_interval_zero() -> anyhow::Result<()> {
+    reset()?;
+
+    // Value of 0, but interval of 100ms
+    let it_value = libc::timeval {
+        tv_sec: 0,
+        tv_usec: 0,
+    };
+    let it_interval = libc::timeval {
+        tv_sec: 0,
+        tv_usec: 100_000,
+    };
+    setitimer(
+        libc::ITIMER_REAL,
+        &libc::itimerval {
+            it_value,
+            it_interval,
+        },
+    )?;
+    // The man page isn't clear about what should happen here,
+    // but experimentally the timer is completely disabled, even though we
+    // specified a non-zero interval.
+    ensure_ord!(getitimer(libc::ITIMER_REAL)?, ==, ITimer{value: TimeVal::zero(), interval: TimeVal::zero()});
```

Comment:
Maybe check that `SIGNAL_CTR` is still zero (I would expect it to be, but just to make sure)?

Reply:
Done